### PR TITLE
test(components/indicators): add wait a11y tests

### PR DESCRIPTION
**Pull request by Blackbaud-TrevorBurch**
[AB#2340629](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2340629)

### CODE REVIEW

File: libs/components/indicators/src/lib/modules/wait/wait.service.spec.ts [hunk 1]
```diff
@@ -88,213 +91,293 @@ describe('Wait service', () => {
...
-    subject.next('A');
-    tick();
-    applicationRef.tick();
-    verifyBlockingPageWaitExists(true);
-    subject.error('error');
-    tick();
-    applicationRef.tick();
-    verifyBlockingPageWaitExists(false);
-  }));
+  describe('accessibility', () => {
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/indicators/src/lib/modules/wait/wait.service.spec.ts:308]: Here down is new.

File: libs/components/indicators/src/lib/modules/wait/wait.component.spec.ts [hunk 2]
```diff
@@ -36,504 +36,582 @@ describe('Wait component', () => {
...
-    fixture.detectChanges();
+      expect(getAriaLabel()).toBe('Page loading.');
+    });
 
-    expect(document.activeElement).not.toBe(button);
+    /**
+     * NOTE: The `region` rule is turned off as our karma tests do not set up regions within the `body`.
+     */
+
+    it('should be accessible when using a full page blocking wait', async () => {
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/indicators/src/lib/modules/wait/wait.component.spec.ts:553]: Here down is new.

File: libs/components/indicators/src/lib/modules/wait/wait.service.spec.ts [hunk 3]
```diff
@@ -18,10 +19,12 @@ describe('Wait service', () => {
   const pageNonBlockingSelector =
     '.sky-wait-page .sky-wait-mask-loading-fixed.sky-wait-mask-loading-non-blocking';
 
-  function beginBlockingPageWait(): void {
+  function beginBlockingPageWait(async = false): void {
     waitSvc.beginBlockingPageWait();
-    tick();
-    applicationRef.tick();
+    if (!async) {
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/indicators/src/lib/modules/wait/wait.service.spec.ts:24]: Made a small tweak here to allow for calling in an `async` spec.

File: libs/components/indicators/src/lib/modules/wait/wait.service.spec.ts [hunk 4]
```diff
@@ -30,10 +33,12 @@ describe('Wait service', () => {
     applicationRef.tick();
   }
 
-  function beginNonBlockingPageWait(): void {
+  function beginNonBlockingPageWait(async = false): void {
     waitSvc.beginNonBlockingPageWait();
-    tick();
-    applicationRef.tick();
+    if (!async) {
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/indicators/src/lib/modules/wait/wait.service.spec.ts:38]: Made a small tweak here to allow for calling in an `async` spec.

File: libs/components/indicators/src/lib/modules/wait/wait.service.spec.ts [hunk 5]
```diff
@@ -76,9 +81,7 @@ describe('Wait service', () => {
     TestBed.configureTestingModule({
       imports: [SkyWaitFixturesModule],
     });
-  });
 
-  beforeEach(() => {
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/indicators/src/lib/modules/wait/wait.service.spec.ts:81]: We had 2 `beforeEach` calls for some reason - it works with them combined so I just cleaned it up.

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/902/storybook/)

Component Storybooks:

- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/902/storybooks/ag-grid-storybook/)
- [indicators-storybook](https://blackbaud.github.io/skyux-pr-preview/902/storybooks/indicators-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/902/storybooks/layout-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/902/storybooks/modals-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/902/storybooks/pages-storybook/)
- [popovers-storybook](https://blackbaud.github.io/skyux-pr-preview/902/storybooks/popovers-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/902/storybooks/split-view-storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/902/storybooks/tabs-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/902/storybooks/theme-storybook/)
- [tiles-storybook](https://blackbaud.github.io/skyux-pr-preview/902/storybooks/tiles-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/902/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/902/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/902/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/902/playground/)

**Review by johnhwhite — APPROVED** [review]

**Pull request merged**